src/actions/actions.js

```diff
@@ -0,0 +1,55 @@
+import { fetchNews } from '../api';
+
+export const ActionTypes = {
+  FETCH_NEWS_BEGIN: 'FETCH_NEWS_BEGIN',
+  FETCH_NEWS_SUCCESS: 'FETCH_NEWS_SUCCESS',
+  FETCH_NEWS_ERROR: 'FETCH_NEWS_ERROR',
+  INVALIDATE_NEWS: 'INVALIDATE',
+};
+
+export function invalidateNews() {
+  return {
+    type: ActionTypes.INVALIDATE_NEWS,
+  };
+}
+
+export function fetchNewsBegin() {
+  return {
+    type: ActionTypes.FETCH_NEWS_BEGIN,
+  };
+}
+
+export function fetchNewsSuccess(articles, page) {
+  return {
+    type: ActionTypes.FETCH_NEWS_SUCCESS,
+    payload: {
+      articles,
+      page,
+    },
+  };
+}
+
+export function fetchNewsError(error) {
+  return {
+    type: ActionTypes.FETCH_NEWS_ERROR,
+    payload: {
+      error,
+    },
+    error: true,
+  };
+}
+
+export function loadArticles() {
+  return (dispatch, getState) => {
+    try {
+      dispatch(fetchNewsBegin());
+
+      const page = getState().valid ? getState().page + 1 : 0;
```

Comment:
extract code and reuse `state`
```
const state = getState();
const { valid } = state;
```

it would be nice to have selector `getNextPageIndex(state)` that would calculate will it return page +1 or 0